builders/build/eth-api/libraries/viem.md

```diff
@@ -0,0 +1,416 @@
+---
+title: How to use viem Ethereum Library
+description: Check out this tutorial to learn how to use the viem TypeScript interface for Ethereum to send transactions and deploy Solidity smart contracts to Moonbeam.
+---
+
+# viem TypeScript Ethereum Library
+
+## Introduction {: #introduction }
+
+[viem](https://viem.sh/){target=_blank} is a modular TypeScript library that allows developers to interact with abstractions over the JSON-RPC API, making it easy to interact with Ethereum nodes. Since Moonbeam has an Ethereum-like API available that is fully compatible with Ethereum-style JSON RPC invocations, developers can leverage this compatibility to interact with Moonbeam nodes. For more information on viem, check out their [documentation site](https://viem.sh/docs/getting-started.html){target=_blank}.
+
+In this guide, you'll learn how to use viem to send a transaction and deploy a contract on the Moonbase Alpha TestNet. This guide can be adapted for [Moonbeam](/builders/get-started/networks/moonbeam/){target=_blank}, [Moonriver](/builders/get-started/networks/moonriver/){target=_blank}, or a [Moonbeam development node](/builders/get-started/networks/moonbeam-dev/){target=_blank}.
+
+## Checking Prerequisites {: #checking-prerequisites }
+
+For the examples in this guide, you will need to have the following:
+
+- An account with funds.
+  --8<-- 'text/faucet/faucet-list-item.md'
+- 
+  --8<-- 'text/common/endpoint-examples-list-item.md'
+
+!!! note
+    --8<-- 'text/common/assumes-mac-or-ubuntu-env.md'
+
+## Installing viem {: #installing-viem }
+
+To get started, you'll need to create a basic TypeScript project. First, create a directory to store all of the files you'll be creating throughout this guide, and initialize the project with the following command:
```

Comment:
I think you should specify what version of node you're using. I tried out the tutorial with v18.18.0